sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -98,16 +103,22 @@ private BatchSpanProcessor(
       int maxQueueSize,
       int maxExportBatchSize,
       int exporterTimeoutMillis) {
-    this.worker =
-        new Worker(
-            spanExporter,
-            scheduleDelayMillis,
-            maxQueueSize,
-            maxExportBatchSize,
-            exporterTimeoutMillis);
-    this.workerThread = new DaemonThreadFactory(WORKER_THREAD_NAME).newThread(worker);
-    this.workerThread.start();
+    this.worker = new Worker(spanExporter, maxQueueSize, maxExportBatchSize, exporterTimeoutMillis);
     this.sampled = sampled;
+    scheduler = Executors.newScheduledThreadPool(1, new DaemonThreadFactory(WORKER_THREAD_NAME));
+    scheduledFutureWithFixedDelay =
+        scheduler.scheduleWithFixedDelay(worker, 0, scheduleDelayMillis, TimeUnit.MILLISECONDS);
```

Comment:
I don't think you want to schedule an export immediately on startup, do you? I'd recommend delaying by the specified delay before the first export.